app/consumer/genesis.go

```diff
@@ -103,6 +95,57 @@ func transformToNew(jsonRaw []byte, ctx client.Context) (json.RawMessage, error)
 	return newJson, nil
 }
 
+// Remove a parameter from a JSON object
+func removeParameterFromParams(params json.RawMessage, param string) (json.RawMessage, error) {
+	paramsMap := map[string]json.RawMessage{}
+	if err := json.Unmarshal(params, &paramsMap); err != nil {
+		return nil, fmt.Errorf("unmarshalling 'params' failed: %v", err)
+	}
+	_, exists := paramsMap[param]
+	if exists {
+		delete(paramsMap, param)
+	}
+	return json.Marshal(paramsMap)
+}
+
+// Transformation of consumer genesis content as it is exported by current provider version
+// to a format supported by consumer version v5.x
+func transformToV5(jsonRaw []byte, ctx client.Context) (json.RawMessage, error) {
```

Comment:
Clarify this comment. The `consumer_id` needs to be removed for consumer chains with either SDK v0.47 and ICS < v4.5.0 or SDK v0.50 and ICS < v6.2.0

Reply:
Note that it's likelt that the changes in v4.5.0 also need to be backported to v5 because dungeon chain uses them.